crates/crypto/src/secp256k1.rs

```diff
@@ -11,6 +11,37 @@ use k256::{
 
 pub struct Secp256k1;
 
+impl Secp256k1 {
+    pub fn extract_public_key(jwk: &Jwk) -> Result<Vec<u8>, CryptoError> {
+        let decoded_x = general_purpose::URL_SAFE_NO_PAD.decode(&jwk.x)?;
+        let decoded_y = general_purpose::URL_SAFE_NO_PAD.decode(
+            jwk.y
+                .as_ref()
+                .ok_or(CryptoError::PublicKeyFailure("missing y".to_string()))?,
+        )?;
+
+        let mut pk_bytes = Vec::with_capacity(1 + decoded_x.len() + decoded_y.len());
+        pk_bytes.push(0x04); // Uncompressed point indicator
```

Comment:
For sure I'll add a few more words there. here's some more context for the cryptographically curious:

sep256k1 public keys are composed of an `x` and `y` coordinate on an elliptic curve. Since elliptic curves are symmetrical, you can omit the `y` coordinate, and just have a single bit that notes if y is positive or negative. This is a "compressed" public key which is 33 bytes long. An "uncompressed" public key has the full y coordinate, making the public key 65 bytes long. To denote that the public key is uncompressed, it's standard to prefix it with the byte `0x04`.

When you're extremely space contrained, such as on bitcoin, shaving off 32 bytes can make a real difference.

This [blog post](https://davidederosa.com/basic-blockchain-programming/elliptic-curve-keys/#:~:text=Public%20keys%20are%2064%20bytes,is%20based%20on%20modular%20arithmetic.) has a pretty friendly explanation.

For an even deeper dive, I'd recommend [Chapter 3 of Jimmy Song's Programming Bitcoin](https://github.com/jimmysong/programmingbitcoin/blob/master/ch03.asciidoc).